Sources/ParseLiveQuery/Client.swift

```diff
@@ -149,11 +152,11 @@ extension Client {
             )
             subscriptions.append(subscriptionRecord)
 
-            if socket == nil {
-                if !disconnected {
-                    reconnect()
-                }
-            } else {
+            if socket == nil && disconnected{
```

Comment:
This could be simplified into this, and probably will be able to get rid of another boolean variable at all.
```Swift
if socket?.readyState == .OPEN {
  sendOperationAsync(.Subscribe(requestId: subscriptionRecord.requestId, query: query))
} else if !disconnected && socket?.readyState != .CONNECTING {
  reconnect()
}
```